perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java

```diff
@@ -3042,6 +3059,18 @@ public void deleteAttribute(PerunSession sess, AttributeDefinition attributeDefi
 
 	}
 
+	@Override
+	public List<Attribute> getRequiredAttributes(PerunSession sess, List<Service> services, Resource resource, Group group, boolean withGroupAttributes) throws GroupResourceMismatchException {
+		this.checkGroupIsFromTheSameVoLikeResource(sess, group, resource);
+
+		List<Attribute> attributes = new ArrayList<>();
+		attributes.addAll(getAttributesManagerImpl().getRequiredAttributes(sess, services, resource, group));
```

Comment:
You can use parameterized constructor call instead of addAll()

Reply:
Done.